File: Frontend/mobile/lib/models/notification.dart
class Notification {
  final int id;
  final String title;
  final String message;
  final String type;
  final bool isRead;
  final int? pfeListingId;
  final int? applicationId;
  final DateTime createdAt;

  Notification({
    required this.id,
    required this.title,
    required this.message,
    required this.type,
    required this.isRead,
    this.pfeListingId,
    this.applicationId,
    required this.createdAt,
  });

  factory Notification.fromJson(Map<String, dynamic> json) {
    return Notification(
      id: json['id'] as int,
      title: json['title'] as String,
      message: json['message'] as String,
      type: json['type'] as String,
      isRead: json['is_read'] as bool,
      pfeListingId: json['pfe_listing_id'] as int?,
      applicationId: json['application_id'] as int?,
      createdAt: DateTime.parse(json['created_at'] as String),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'title': title,
      'message': message,
      'type': type,
      'is_read': isRead,
      'pfe_listing_id': pfeListingId,
      'application_id': applicationId,
      'created_at': createdAt.toIso8601String(),
    };
  }
}

File: Backend/app/models/notification.py
from sqlalchemy import Column, Integer, String, Boolean, DateTime, ForeignKey, Text, Enum
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
import enum
from app.db.database import Base


class NotificationType(str, enum.Enum):
    NEW_APPLICATION = "new_application"
    APPLICATION_STATUS = "application_status"
    PFE_UPDATE = "pfe_update"
    SYSTEM = "system"


class Notification(Base):
    """Notification model for user notifications"""
    __tablename__ = "notifications"
    
    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id", ondelete="CASCADE"), nullable=False)
    title = Column(String(255), nullable=False)
    message = Column(Text, nullable=False)
    type = Column(Enum(NotificationType), default=NotificationType.SYSTEM, nullable=False)
    is_read = Column(Boolean, default=False)
    
    # Optional reference to related entities
    pfe_listing_id = Column(Integer, ForeignKey("pfe_listings.id", ondelete="SET NULL"), nullable=True)
    application_id = Column(Integer, ForeignKey("applications.id", ondelete="SET NULL"), nullable=True)
    
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    
    # Relationships
    user = relationship("User", backref="notifications")
    pfe_listing = relationship("PFEListing", backref="notifications")
    application = relationship("Application", backref="notifications")
